evergreen.yml

```diff
@@ -795,6 +985,14 @@ tasks:
       - func: "compile release"
       - func: "mciuploads release build and symbols"
 
+  - name: compile-macos
+    depends_on:
+    commands:
+      - func: "install iODBC"
+      - func: "set packages version"
+      - func: "compile macos release"
+      - func: "mciuploads release build and symbols"
```

Comment:
Instead of using a compile-macos, we can extend what we are currently doing for compile:

```
      - name: compile
        depends_on:
        commands:
          - func: "install unix odbc"
            variants: [ubuntu2204]
          - func: "install iODBC"
            variants: [macos]
          - func: "set packages version"
          - func: "compile release"
          - func: "build msi"
            variants: [windows-64]
          - func: "mciuploads release artifacts"
   ```